app/assets/javascripts/settings/ldap-conn-test.js

```diff
@@ -0,0 +1,78 @@
+$(function () {
+  function NoneEmpty (arr) {
+    for (var i = 0; i < arr.length; i++) {
+      if (arr[i] === '') return false
+    }
+    return true
+  };
+
+  function RenderValidity (msg, isValid) {
+    if (isValid) {
+      $('#ldap_conn_save').removeProp('disabled')
+      $('#ldap_conn_message').attr('class', 'ldap-conn-message-valid')
```

Comment:
Instead of `attr`, prefer `addClass`.

Reply:
'attr' was initially chosen due to its conciseness - no need to remove and add classes as it overwrites the current class.  In hindsight, it's obvious that it it would prevent the use of additional classes as they would get replaced.  Fixed using your suggestion now.